src/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -398,6 +400,31 @@ public void Timeout_TooShort_AllPendingOperationsCanceled()
             }
         }
 
+        [Fact]
+        public async Task SendAsync_UserAgent_CorrectlyWritten()
+        {
+            string userAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.18 Safari/537.36";
+            bool connectionAccepted = false;
+
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
+            {
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
+                {
+                    var message = new HttpRequestMessage(HttpMethod.Post, uri);
+                    message.Headers.TryAddWithoutValidation("User-Agent", userAgent);
+                    await client.SendAsync(message).ConfigureAwait(false);
+                }
+            }, server => server.AcceptConnectionAsync(async connection =>
+            {
+                connectionAccepted = true;
+                List<string> headers = await connection.ReadRequestHeaderAndSendResponseAsync();
+                Assert.Contains($"User-Agent: {userAgent}", headers);
+            }));
+
+            Assert.True(connectionAccepted);
```

Comment:
The connection accepted isn't necessary anymore.  If it's not accepted, the client request will fail.